react/src/base/inputs/SprkCheckbox/SprkCheckboxGroup/SprkCheckboxGroup.js

```diff
@@ -11,51 +10,61 @@ const SprkCheckboxGroup = (props) => {
     idString,
     additionalClasses,
     analyticsString,
+    ariaDescribedBy,
+    ...rest
   } = props;
 
-  let errorId = null;
-  let helperId = null;
-  let hasErrorContainer = false;
-  let hasHelperText = false;
-
+  let errorId;
+  let helperId;
+  let inputAriaDescribedByArray = [];
+  if (ariaDescribedBy) {
+    inputAriaDescribedByArray = ariaDescribedBy.split(' ');
+  }
   const childrenArray = React.Children.toArray(children);
-  // Map through children, assign SprkErrorContainer with id/uniqueId
-  // Assign errorId which will determine the ariaDescribedBy of SprkCheckboxItem
+  // Get the errorId and helperId.
+  // If they don't exist in the ariaDescribedBy, add them.
   const elementsToProcess = childrenArray.map((element) => {
-    if (element.type.name === 'SprkErrorContainer') {
-      hasErrorContainer = true;
-      errorId = element.props.id || uniqueId('sprk-error-container-');
-      return React.cloneElement(element, { id: errorId });
+    if (
+      element.type.name === 'SprkFieldError' ||
+      element.type.name === 'SprkErrorContainer'
+    ) {
+      errorId = element.props.id;
+      if (!inputAriaDescribedByArray.includes(errorId)) {
+        inputAriaDescribedByArray.push(errorId);
+      }
+      return React.cloneElement(element);
     }
 
     if (element.type.name === 'SprkHelperText') {
-      hasHelperText = true;
-      helperId = element.props.id || uniqueId('sprk-helper-text-');
-      return React.cloneElement(element, { id: helperId });
+      helperId = element.props.id;
+      if (!inputAriaDescribedByArray.includes(helperId)) {
+        inputAriaDescribedByArray.push(helperId);
+      }
+      return React.cloneElement(element);
```

Comment:
So what's the different between this (line 38-43) and the one before this (27-35)
They both add the ID if it isn't there and return the element. Seems redundant in my head because you can grab its id by by adding another `|| element.typename==='SprkHelperText` above

Reply:
If I add it to the one above then if there is a `SprkFieldError` and `SprkHelperText` the function will stop evaluating because it will hit a TRUE statement and `SprkHelperText` won't get looked at. We need to have them separate because we need to check for both, not one or the other.